src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php

```diff
@@ -675,4 +676,28 @@ public function testDoNotDiscardReturnTypeErrorWhenWriterMethodIsMisconfigured()
 
         $this->propertyAccessor->setValue($object, 'name', 'foo');
     }
+
+    public function testWriteToSingularPropertyWhilePluralOneExists()
+    {
+        $object = new TestSingularAndPluralProps();
+
+        if ($this->propertyAccessor->isWritable($object, 'email')) {
+            $this->propertyAccessor->setValue($object, 'email', 'test@email.com');
+        }
+
+        self::assertEquals('test@email.com', $object->getEmail());
+        self::assertEmpty($object->getEmails());
+    }
+
+    public function testWriteToPluralPropertyWhileSingularOneExists()
+    {
+        $object = new TestSingularAndPluralProps();
+
+        if ($this->propertyAccessor->isWritable($object, 'emails')) {
```

Comment:
instead of having this if, I would have `$this->assertTrue($this->propertyAccessor->isWritable($object, 'emails')`. We want the property to be writable in this test (and same thing for the previous test).

and maybe we should even write without guarding the call at all (so that if the property is not writable, we get the detailed error message in the failure)

Reply:
`isWritable` is not a guard here but necessary condition for reproducing this issue.
`isWritable` internally caches the info to writePropertyCache with 3rd argument as array `getWriteAccessInfo(\get_class($object), $property, array());` which is used by findAdderAndRemover.
Do you still think assertTrue is needed for isWritable?

Reply:
If I don't call isWritable - then `getWriteAccessInfo` will be called by `writeProperty` with 3rd argument as string: `$this->getWriteAccessInfo(\get_class($object), $property, $value);` and findAdderAndRemover will not be called

Reply:
maybe this would be better?
```
$this->propertyAccessor->isWritable($object, 'email'); //cache access info
$this->propertyAccessor->setValue($object, 'email', 'test@email.com');
```